### PR TITLE
fix(provider-code): newlines preserved in UI

### LINKED ISSUE #3174: Bug: Providers code examples have inconsistent spacing
### Steps to reproduce the problem

1. Go to the [Qiskit.org Providers page](https://qiskit.org/providers)
2. Scroll down to the Providers widget where you see code examples
4. See the code examples on the right

### What is the current behavior?

We're missing the empty spaces between `<pre>` lines of code, and `# Code comment` lines of code

### What is the expected behavior?

There should be empty spaces between `<pre>` lines of code, and `# Code comment` lines of code

## Providers page code examples
Screenshot 2023-05-02 at 7 17 12 AM (2)

### Additional information
For the homepage, the spacing is inconsistent — sometimes a blank line shows empty spacing, and at other times it doesn't.  If the user clicks around to see different providers, then goes back to the `Qiskit (with built-in simulator)` examples, the spacing will show, partially show, or not at all.

👇🏼 

https://user-images.githubusercontent.com/6276074/235701249-01da6abe-3757-463f-bbf3-7df67055dc51.mov

### PR DESCRIPTION
**Pull request by jackoconnordev**
## Changes

Newlines in the text provided to provider code blocks will now appear in the UI.

Also added (I believe) missing newlines in the source text for some provider code. All comments should have a newline above them right?

Fixes #3174 

## Implementation details

Newline characters aren't preserved by \<pre\> tag but spaces are. The laptop code block on the homepage's
text is hardcoded on the homepage and it includes space characters in the blank lines.

The provider text used newline characters on blank lines but for some reason \<pre\> tags don't preserve that.
Fix was to just use spaces instead.

~~The previous attempt to fix newlines not appearing ended up overpowered using `computed`
so I've streamlined it a bit since this isn't a dynamic component.~~ It does need `computed`, it is a reactive element on the home page.

#### Before

image

#### After

image

### CODE REVIEW

**Comment by jackoconnordev** [comment]
Can't find the option to convert back to draft sorry so I'll close and open a new one when found. The fix was only working on dev server for me, I checked previewing a build and the whitespace disappears again. It's possible that Nuxt build removes the empty lines from the source text? I will attempt to verify.

The npm run generate step of the build action still can't find the airtable key for me for some reason so there's no live preview of the issue. Not sure what I'd doing wrong with that, if anything.

File: components/Ui/UiCodeSnippet.vue [hunk 1]
```diff
@@ -34,17 +34,12 @@ interface Props {
...
-    return line;
-  });
-  return codeLinesWithEmptyLines;
-});
+const codeLines = props.code.split("\n");
+for (const line in codeLines) {
+  if (codeLines[line] === "") {
+    codeLines[line] = " ";
+  }
+}
```
Comment (techtolentino) [inline components/Ui/UiCodeSnippet.vue:42]: is this working on your side, @jackoconnordev?


Uploading Screen Recording 2023-07-24 at 12.21.04 PM.mov…

**Review by techtolentino — CHANGES_REQUESTED** [review]
Hi @jackoconnordev thanks for the PR

Please see if this is causing any issues. When I checkout the PR locally, I'm not able to see new code snippets per provider in the homepage quick start switcher.

**Comment by jackoconnordev** [comment]
I wrongly suggested there was no need for the UICodeSnippet text to be reactive. The text variable is now computed at each rerender again and should work as intended. Thanks for spotting my mistake @techtolentino.

(Please excuse the terrible quality 720p was just over the 10mb limit. The text changing on the homepage should still be clear in the second half of the video.)

https://github.com/Qiskit/qiskit.org/assets/134723582/aaa97634-6911-40c8-94ed-45c6be93e7cd

**Comment by techtolentino** [comment]
@jackoconnordev - since you also approve of https://github.com/Qiskit/qiskit.org/pull/3450, we can close this.

Thank you for your involvement.

Let's wait on @y4izus to also review, then we can merge that PR...

**Pull request closed without merging**